narayana/scripts/hudson/bm.sh

```diff
@@ -19,10 +19,18 @@ function cmp_narayana {
 
 function runs {
   echo -e "threads\tmaxIO\tbf/s\tAIO\ttput\t\tchange" > results.txt
+  PROGRESS=0
+  INC=100
+  LOOP_ITERATION=16
   for threads in 1 24 240 1600; do
     for bf in 300 2000 4000 7812; do
+      PROGRESS=`expr $PROGRESS + $INC`
```

Comment:
These changes look like overkill if your only goal is to be able to debug its' progress. Adding the following line will provide you with the same data and will be much easier for engineers to maintain:

`echo "$(($(date +%s%N)/1000000)): $threads threads and $bf buffer flushes"`

Reply:
But personally I would just change the job that calls the script, that way you don't need to make any changes to the code base:

`PS4='+ $(date -Ins): ' sh -x ./narayana/scripts/hudson/bm.sh`

This way we can perform debugging on demand.

Reply:
Logging the date for each iteration would be more informative and will let us know the progress, so the first options that you suggested would be good fit here.

Reply:
But this change is only to debug hangs and we only need the extra output on demand so why not use the PS4 prompt instead (which will aid debugging of all parts of the script rather than this single function). If we change the code base every time we need extra information then its' maintenance will become unwieldy.

Reply:
Okay Mike, I will close this PR and update the job.